src/Revisions/RevisionRepository.php

```diff
@@ -36,6 +36,11 @@ public function findWorkingCopyByKey($key)
 
     public function save(RevisionContract $copy)
     {
+        if ($copy instanceof WorkingCopy) {
+            app('statamic.eloquent.revisions.model')::where('key', $copy->key())
```

Comment:
Hi! I just tested your latest changes, and I believe that this also removes any past revisions with action `revision` as well? I'm thinking that something along the lines of `where(['key' => $key, 'action' => 'working'])` would be more suitable? Please correct me if I'm wrong here. 

And many thanks for all the hard work you're putting in to this, it really is worth a lot!

Reply:
Yes of course - good spot. I've updated it now.